proofread fault_diagnosis_and_analysis (#2526)
juicedata/juicefs: docs/en/administration/fault_diagnosis_and_analysis.md

## Client log

> JuiceFS client will output logs for troubleshooting during operation. Log levels in order of fatality are: DEBUG, INFO, WARNING, ERROR and FATAL. Since DEBUG logs are not printed by default, you need to explicitly enable it if you need, e.g. by adding the `--debug` option, when running the JuiceFS client, 

Reviewer: ```suggestion
JuiceFS client will output logs for troubleshooting while running. The level of logs in terms of fatality follows DEBUG < INFO < WARNING < ERROR < FATAL. Since DEBUG logs are not printed by default, you need to explicitly enable it if needed, e.g. by adding the `--debug` option when running the JuiceFS client. 
```
- Maybe "while running" is better than "during operation"?

> Different JuiceFS clients print logs in different ways, described as follows.

Reviewer: ```suggestion
Different JuiceFS clients print logs in different ways, which are described as follows.
```

### Mount point

> When a JuiceFS file system is mounted with the [`-d` option](../reference/command_reference.md#juicefs-mount) (indicating running in the background), it will print logs to the system log file and local log file simultaneously. Depending on the user JuiceFS client running as, the path of the local log files are slightly different. The local log file locates at `/var/log/juicefs.log` when running as root, while `$HOME/.juicefs/juicefs.log` for non-root users. For details. See [`--log` option](../reference/command_reference.md#juicefs-mount).

Reviewer 2: `-d` stands for `--background`, expand for clarification:

When a JuiceFS file system is mounted with the [`-d`/`--background` option](../reference/command_reference.md#juicefs-mount), it will print logs to the system log file and local log file simultaneously.

local log file path is `/var/log/juicefs.log` when running as root, `$HOME/.juicefs/juicefs.log` for non-root users.

> Shall we standardize the phrases

no need to toil just for this purpose, but better ditch the "please" in "please refer to".

> Different commands here to get local logs for different operating systems.

Reviewer: ```suggestion
Depending on the operating system, there are different commands that you can use to obtain system logs or read local log files directly. 
```

### Kubernetes CSI Driver

> Depending on the version of the JuiceFS CSI Driver you are using, there will be different ways to retrieve logs. For details, please refer to [CSI Driver documentation](https://juicefs.com/docs/csi/troubleshooting).

Reviewer: ```suggestion
Depending on the version of the JuiceFS CSI Driver, there are different ways to retrieve logs. Please refer to [CSI Driver documentation](https://juicefs.com/docs/csi/troubleshooting) for details.
```
- It is a statement, so I think we can use the present tense here.

Reviewer 2: can we ditch the "please" in all future work? not just to make things shorter, tech docs (that I've seen) always use neutral tone.

### S3 Gateway

> The S3 gateway can only run in the foreground, so client logs are output directly to the terminal. If you deploys the S3 gateway in Kubernetes, you can get logs from the corresponding pods.

Reviewer: ```suggestion
The S3 gateway can only run in the foreground, so client logs are printed directly in the terminal. If the S3 gateway is deployed in Kubernetes, you can retrieve the corresponding pod logs.
```
Based on the Chinese document, if I understand correctly, the last sentence says that "if ...., we need to check the pod logs". But why we NEED to check the pod logs? Are the pod logs the prerequisites for other logs? Or it is just "get/retrieve the pod logs"? Sorry for the newbie question. :P

Reviewer 2: > so client logs are printed directly in the terminal.

"printed to terminal" and "output to console" are both ok, they are used quite often.

and for the kubernetes part, I think it'll be fine with:

If deployed in Kubernetes, view the pod logs.

Author: @AIXjing It's NO NEED to check the pods logs. I guess the author of the Chinese doc was expressing "you need to check logs of the corresponding pods (rather than somewhere else) if you need to"

### Hadoop Java SDK

> The JuiceFS client logs will be mixed into the logs of process which uses JuiceFS Hadoop Java SDK, e.g. Spark executor. You need to use keywords, e.g. `juicefs` which is case-insensitive, to filter out the logs you want.

Reviewer: ```suggestion
The JuiceFS client logs will be mixed into the logs of processes using JuiceFS Hadoop Java SDK, e.g. Spark executor. Thus, you need to use keywords, e.g. `juicefs` (case-insensitive), to filter out the logs you do not want.
```

## Access log

> Each JuiceFS client has an access log that records all operations on the file system in detail, such as operation type, user ID, group ID, file inodes and time cost. Access logs can be used for various purposes such as performance analysis, auditing, troubleshooting.

Reviewer: ```suggestion
Each JuiceFS client has an access log that records all operations on the file system in detail, such as operation type, user ID, group ID, file inodes and time cost. Access logs can be used for various purposes such as performance analysis, auditing, and troubleshooting.
```

- `[uid:0,gid:0,pid:4403]`: User ID, group ID, process ID of the current operation
- `write`: Operation type
- `(17669,8666,4993160)`: The input parameters of the current operation type. For example, the input parameters of the `write` operation in the example are the inode of the written file, the size of the written data, and the offset of the written file. Different operation types have different parameters. For details, please refer to the [`vfs.go`](https://github.com/juicedata/juicefs/blob/main/pkg/vfs/vfs.go) file.

> - `OK`: Indicate the current operation is successful or not. If it is unsuccessful, specific failure information will be output.

Reviewer: ```suggestion
- `OK`: Indicate whether the current operation is successful or not. If it is unsuccessful, specific failure information will be output.
```

Reviewer 2: specific failure information will be printed.

Author: I didn't replace `output` with `print`, even though I would like to. `output` is correct but a little bit Chinglish.

> - `<0.000010>`: The time (in seconds) that the current operation cost

Reviewer: ```suggestion
- `<0.000010>`: The time (in seconds) that the current operation takes
```

> You can debug and analyze performance issues with access log, or try using `juicefs profile <mount-point>` to see real-time statistics. Run `juicefs profile -h` or refer to [here](../benchmark/operations_profiling.md) to learn more about this subcommand.

Reviewer: ```suggestion
You can debug and analyze performance issues with access log, or try using `juicefs profile <mount-point>` to see real-time statistics. Run `juicefs profile -h` or refer to [Operations Profiling](../benchmark/operations_profiling.md) for details.
```

```bash
$ cat /jfs/.accesslog
```

```output
2021.01.15 08:26:11.003330 [uid:0,gid:0,pid:4403] write (17669,8666,4993160): OK <0.000010>
2021.01.15 08:26:11.003473 [uid:0,gid:0,pid:4403] write (17675,198,997439): OK <0.000014>
2021.01.15 08:26:11.003616 [uid:0,gid:0,pid:4403] write (17666,390,951582): OK <0.000006>
```

### Kubernetes CSI Driver

> Please refer to [CSI Driver documentation](https://juicefs.com/docs/csi/troubleshooting) and find the mount pod according or CSI driver pod to the version of JuiceFS CSI Driver you are using. You can find the `.accesslog` file in the root directory of the JuiceFS file system mount point in the pod. The mount point path in the pod is `/jfs/<pv_volumeHandle>`, assuming the name of the mount pod is `juicefs-1.2.3.4-pvc-d4b8fb4f-2c0b-48e8-a2dc-530799435373`, `<pv_volumeHandle>` is `pvc-d4b8fb4f-2c0b-48e8-a2dc-530799435373`, you can use the following command to view:

Reviewer: ```suggestion
Please refer to [CSI Driver documentation](https://juicefs.com/docs/csi/troubleshooting) to find the mount pod or CSI driver pod depending on the version of JuiceFS CSI Driver you are using, and the `.accesslog` file can be viewed in the root directory of the JuiceFS file system mount point in the pod. The mount point path in the pod is `/jfs/<pv_volumeHandle>`. Assuming there is a mount pod named as `juicefs-1.2.3.4-pvc-d4b8fb4f-2c0b-48e8-a2dc-530799435373`, in which `pvc-d4b8fb4f-2c0b-48e8-a2dc-530799435373` is  `<pv_volumeHandle>`, you can then use the following command to view the `.accesslog` file:
```

By default, JuiceFS clients will listen to a TCP port locally via [pprof](https://pkg.go.dev/net/http/pprof) to get runtime information such as Goroutine stack information, CPU performance statistics, memory allocation statistics. You can see the specific port number that the current JuiceFS client is listening on by using the system command (e.g. `lsof`):

:::note

> If you mount JuiceFS as the root user, you need to add `sudo` before the `lsof` command.

Reviewer: ```suggestion
If JuiceFS is mounted with the root user, you need to add `sudo` before the `lsof` command.
```
Try to avoid first-person pronouns. :)

Author: Would it be better to replace `with` with `by`?

:::

```bash
$ lsof -i -nP | grep LISTEN | grep juicefs
```

```output
juicefs   32666 user    8u  IPv4 0x44992f0610d9870b      0t0  TCP 127.0.0.1:6061 (LISTEN)
juicefs   32666 user    9u  IPv4 0x44992f0619bf91cb      0t0  TCP 127.0.0.1:6071 (LISTEN)
juicefs   32666 user   15u  IPv4 0x44992f062886fc5b      0t0  TCP 127.0.0.1:9567 (LISTEN)
```

> By default, pprof listens on port numbers ranging from 6060 to 6099, that's why the actual port number in the above example is 6061. Once you get the listening port number, you can view all the available runtime information by accessing `http://localhost:<port>/debug/pprof`, and some important runtime information is as follows:

Reviewer: ```suggestion
By default, pprof listens on port numbers ranging from 6060 to 6099. That's why the actual port number in the above example is 6061. Once you get the listening port number, you can view all the available runtime information by accessing `http://localhost:<port>/debug/pprof`, and some important runtime information will be shown as follows:
```
